Code/GraphMol/MolOps.cpp

```diff
@@ -1168,5 +1168,105 @@ std::vector<int> hapticBondEndpoints(const Bond *bond) {
   return oats;
 }
 }  // end of namespace details
-};  // end of namespace MolOps
-};  // end of namespace RDKit
+
+void expandAttachmentPoints(RWMol &mol, bool addAsQueries) {
+  for (auto atom : mol.atoms()) {
+    int value;
+    if (atom->getPropIfPresent(common_properties::molAttachPoint, value)) {
+      std::vector<int> tgtVals;
+      if (value == 1 || value == -1) {
+        tgtVals.push_back(1);
+      }
+      if (value == 2 || value == -1) {
+        tgtVals.push_back(2);
+      }
+      if (tgtVals.empty()) {
+        BOOST_LOG(rdWarningLog)
+            << "Invalid value for molAttachPoint: " << value << " on atom "
+            << atom->getIdx() << ". Not expanding this atttachment point."
+            << std::endl;
+        continue;
+      }
+      for (auto tval : tgtVals) {
+        atom->clearProp(common_properties::molAttachPoint);
+        Atom *newAtom = nullptr;
+        if (addAsQueries) {
+          newAtom = new QueryAtom(0);
+          newAtom->setQuery(RDKit::makeAtomNullQuery());
+        } else {
+          newAtom = new Atom(0);
+        }
+        newAtom->setProp(common_properties::_fromAttachPoint, tval);
```

Comment:
would it make sense to decorate these attachment dummies with _AP labels so they match CXSMILES output?
```
dummy_atom->setProp(RDKit::common_properties::atomLabel, "_AP" + std::to_string(i));
```

Also, would it be possible to expose a function that creates a new attachment point atom prior to insertion into the mol? The idea is that the sketcher would re-use that logic when adding attachment points. Something like:
```
std::shared_ptr<RDKit::Atom> make_new_attachment_point(const unsigned int tval);
```

Reply:
It seems like it would be more useable to have `unsigned int addExplicitAttachmentPoint(Atom &atom,unsigned int tVal)`, which would create and add the new dummy, add the bond, and do whatever property adjustments are required on `atom` and then the index of the newly added attachment point?